File: VB/Invoice.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text

Namespace SchedulerAppointmentChildObjects
	Public Class Invoice
		Private id_Renamed As Integer

		Public Property Id() As Integer
			Get
				Return id_Renamed
			End Get
			Set(ByVal value As Integer)
				id_Renamed = value
			End Set
		End Property
		Private description_Renamed As String

		Public Property Description() As String
			Get
				Return description_Renamed
			End Get
			Set(ByVal value As String)
				description_Renamed = value
			End Set
		End Property
		Private price_Renamed As Decimal

		Public Property Price() As Decimal
			Get
				Return price_Renamed
			End Get
			Set(ByVal value As Decimal)
				price_Renamed = value
			End Set
		End Property

		Public Sub New(ByVal id As Integer, ByVal description As String, ByVal price As Decimal)
			Me.Id = id
			Me.Description = description
			Me.Price = price
		End Sub

		' Serialization constructor
		Public Sub New()
			Me.New(0, String.Empty, 0D)

		End Sub
	End Class
End Namespace

File: CS/Invoice.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SchedulerAppointmentChildObjects {
    public class Invoice {
        private int id;

        public int Id {
            get { return id; }
            set { id = value; }
        }
        private string description;

        public string Description {
            get { return description; }
            set { description = value; }
        }
        private decimal price;

        public decimal Price {
            get { return price; }
            set { price = value; }
        }

        public Invoice(int id, string description, decimal price) {
            this.Id = id;
            this.Description = description;
            this.Price = price;
        }

        // Serialization constructor
        public Invoice()
            : this(0, string.Empty, 0m) {

        }
    }
}
